README.md

```diff
@@ -198,54 +213,204 @@ The method `evaluate_quadratic_expression` has the following interface:
 
+To constrain the example, first calculate `g = a * b + (c + d) * e + f` in unconstrained functions.
 ```rust
-let lhs_terms = [[a, b], [d, e]];
-let lhs_flags = [[false, false], [false, true]];
-let rhs_terms = [[c], [f]];
+// First product term a * b
+let t0 = a.__mul(b);
+// Second product term (c + d) * e
+let t1 = (c.__add(d)).__mul(e); 
+let g = t0.__add(t1).__add(f);
+```
+
+
+
+Then define the terms and flags to constrain `a * b + (c + d) * e + f - g = 0` and call `evaluate_quadratic_expression`. 
+```rust
+// (a + 0) * b
+let lhs_terms = [[a, BigNum::new()], [c, d]];
+let lhs_flags = [[false, false], [false, false]];
+// (c + d) * e 
+let rhs_terms = [[b], [e]];
 let rhs_flags = [[false], [false]];
-let add_terms = [g];
-let add_flags = [true];
-BigNum::evaluate_quadratic_expresson(lhs_terms, lhs_flags, rhs_terms, rhs_flags, linear_terms, linear_flags);
+// + f + (-g)
+let add_terms = [f, g];
+let add_flags = [false, true];
+
+BigNum::evaluate_quadratic_expression(lhs_terms, lhs_flags, rhs_terms, rhs_flags, linear_terms, linear_flags);
 ```
 
-##### TODO: Document other available methods
+#### Example usage evaluate_quadratic_expression
 
-#### Deriving BigNumParams parameters: `modulus`, `redc_param`
+In the base field of Ed25519, which is the integers mod $2^{255}-19$, perform simple arithmetic operations `(x1 * x2) + x3` and assert this equals `expected`. 
 
-For common fields, BigNumParams parameters can be pulled from the presets in `bignum/fields/`.
+```rust
+use dep::bignum::BigNum;
+use dep::bignum::fields::ed25519Fq::ED25519_Fq_Params;
+
+// Prime field mod 2^255-19
+type Fq = BigNum<3, 255, ED25519_Fq_Params>;
+
+// Check that (x1 * x2) + x3 equals `expected`
+fn main(x1: Fq, x2: Fq, x3: Fq, expected: Fq) {
+    // Step 1: calculate res = (x1 * x2) + x3 in unconstrained functions
+    let res = x1.__mul(x2).__add(x3);
+
+    // Step 2: Constrain (x1 * x2) + x3 - res == 0 mod 2^255-19
+    // (until now we have value res, but "unchecked")
+
+    // `evaluate_quadratic_expression` takes:
+    // (1) x1, `false` sign flag
+    // (2) x2, `false` sign flag
+    // (3)
+    //  - x3, `false` sign flag
+    //  - res, `true` sign flag
+    // Combines to: (x1 * x2) + x3 + (-res)
+    BigNum::evaluate_quadratic_expression(
+        [[x1]],
+        [[false]],
+        [[x2]],
+        [[false]],
+        [x3, res],
+        [false, true],
+    );
+
+    // Step 3: check res equals `expected`
+    assert(res == expected); // Equality operation on BigNums
+}
+```
 
-For other moduli (e.g. those used in RSA verification), both `modulus` and `redc_param` must be computed and formatted according to the following speficiations:
+See `bignum_test.nr` for more examples.
 
-`modulus` represents the BigNum modulus, encoded as an array of `Field` elements that each encode 120 bits of the modulus. The first array element represents the least significant 120 bits.
+## Custom or predefined parameter set
```

Comment:
Previously I've noticed "parameterset" as one word, and didn't know if it was intentional?